pkg/log/telemetry.go

```diff
@@ -0,0 +1,146 @@
+package log
+
+import (
+	"crypto/sha1"
+	"fmt"
+
+	"github.com/SAP/jenkins-library/pkg/piperenv"
+	"github.com/sirupsen/logrus"
+)
+
+// SWA Reporting
+// 1. Step usage, details, ... -> SWA implementierung
+// 2. Errors -> Hook
+// 3. Notify/Deprecations
+
+// TelemetryBaseData ...
+type TelemetryBaseData struct {
+	Active     bool
+	ActionName string `json:"actionName,omitempty"`
+	EventType  string `json:"eventType,omitempty"`
+	SiteID     string `json:"idsite,omitempty"`
+	URL        string `json:"url,omitempty"`
+
+	GitOwner               string `json:"e_a,omitempty"`
```

Comment:
In order go avoid some possible error cases from the root on I would like to propose to even not have fields for the un-hashed values of owner/repo.